lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -403,30 +405,36 @@ def margins(self, *args, **kw) :
         mx = kw.pop('x', None)
         my = kw.pop('y', None)
         mz = kw.pop('z', None)
-        if len(args) == 1:
+        if not args:
+            pass
+        elif len(args) == 1:
             mx = my = mz = args[0]
         elif len(args) == 2:
-            # Maybe put out a warning because mz is not set?
+            warnings.warn(
+                "Passing exactly two positional arguments to Axes3D.margins "
+                "is deprecated.  If needed, pass them as keyword arguments "
+                "instead", cbook.mplDeprecation)
             mx, my = args
         elif len(args) == 3:
             mx, my, mz = args
         else:
-            raise ValueError("more than three arguments were supplied")
+            raise TypeError(
+                "Axes3D.margins takes at most three positional arguments")
```

Comment:
grrr, misread the code... it doesn't have an else clause... it has a "len(args) > 2" clause that then raises the ValueError, but also skips the case of no positional args. We are fine, then.

As for ValueError vs. TypeError, I really don't have a preference except that the 2d and 3d version be the same.